src/Symfony/Component/Security/Http/AccessToken/OAuth2/Exception/MissingClaimException.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken\OAuth2\Exception;
+
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+/**
+ * This exception is thrown when a claim is missing or invalid.
+ *
+ * @experimental
+ */
+class MissingClaimException extends AuthenticationException
```

Comment:
I'm not sure we need this very specific exception given it's immediately caught where it's throw and wrapped in a `BadCredentialsException`. I suggest to throw `BadCrendentialsException`s with precise messages where needed instead (see https://github.com/symfony/symfony/pull/50695 - I'll finish it at some point so better not continue this way)